### PR TITLE
fix: prevent creation of nightly release with  draf==true

### PR DESCRIPTION
**Pull request by hassec**
Sometimes the nightly pre-release would be in "draft" status on GitHub.   
It's hard to reproduce, but I had the guess that maybe the deletion of
old tag and release on the GitHub side wasn't quite done when the
new release was being created. Of course only speculation...
But, moving the download of the artifacts in between these two steps
seems to have fixed the symptom. 

Given how difficult it is to debug GitHub actions, I'd say let's merge this and keep an eye on it. 
I can spend more time in the case the problem persists.

### CODE REVIEW

**Review by piponazo — APPROVED** [review]
I'm approving but I wonder if you could test this new approach in your own fork for few days before merging it in the main repository. Would that be possible? or tests need to be done in the main repository anyways?

**Comment by hassec** [comment]
@piponazo thanks for having a look, that's actually what I did.   
I triggered the workflow a bunch of times, and about every 2-4 times, I'd be able to reproduce the error.
I had even added more printouts of e.g. each GitHub API response and more but didn't see anything different for the jobs that only released a draft. 

After this change I've triggered O(10) jobs yesterday and wasn't able to reproduce the failure case.

**Comment by piponazo** [comment]
> @piponazo thanks for having a look, that's actually what I did.
> I triggered the workflow a bunch of times, and about every 2-4 times, I'd be able to reproduce the error.
> I had even added more printouts of e.g. each GitHub API response and more but didn't see anything different for the jobs that only released a draft.
> 
> After this change I've triggered O(10) jobs yesterday and wasn't able to reproduce the failure case.

Thanks for the info! Go ahead and merge the PR whenever you want 😉

**Pull request merged**
